source/material_model/equation_of_state/linearized_incompressible.cc

```diff
@@ -0,0 +1,159 @@
+/*
+  Copyright (C) 2011 - 2018 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/material_model/equation_of_state/linearized_incompressible.h>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace EquationOfState
+    {
+      template <int dim>
+      void
+      LinearizedIncompressible<dim>::
+      evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+               const unsigned int q,
+               MaterialModel::EquationOfStateOutputs<dim> &out) const
+      {
+
+        Assert(number_of_compositions+1 >= out.densities.size(),
+               ExcMessage("Error: You are trying to evaluate the equation of state with "
+                          + Utilities::to_string(out.densities.size()-1) +
+                          " compositional fields, which is larger than "
+                          + Utilities::to_string(number_of_compositions) +
+                          ", the number of fields the equation if state was set up with "
+                          "in the 'set_number_of_compositions' function."));
+
+
+        for (unsigned int c=0; c < out.densities.size(); ++c)
+          {
+            out.densities[c] = reference_rho * (1 - thermal_alpha * (in.temperature[q] - reference_T));
+            if (c>0)
+              out.densities[c] += compositional_delta_rhos[c-1];
+
+            out.thermal_expansion_coefficients[c] = thermal_alpha;
+            out.specific_heat[c] = reference_specific_heat;
+            out.compressibilities[c] = 0.0;
+            // Pressure derivative of entropy at the given positions.
```

Comment:
Remove these comments, they belong in the header file for this variable.

Reply:
Oh, can I do that for other material models as well?
I'll get lots of pull requests in!

Reply:
Yes, I think there are a lot of comments that actually should be in the header, and that will hopefully be reduced by deduplicating all the EoS code.